src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs

```diff
@@ -445,6 +445,12 @@ public Task<Block> GetBlockAsync(uint256 hash)
                         res = blockRow.Value;
                 }
 
+                // If searching for genesis block, return it.
+                if (res == null && hash == this.network.GenesisHash)
```

Comment:
Hmm my intention was to implement this in IBlockStore, I guess here is also fine.

Reply:
I might be able to do that, I will take a quick look.